File: goofy-protocol-fis-be/src/main/java/com/masl/goofy_protocol_core/crypto/isolated/symm/SymmCryptoType.java
package com.masl.goofy_protocol_core.crypto.isolated.symm;

import lombok.Getter;

@Getter
public enum SymmCryptoType {
    AES_GCM_128((short)0x0100),
    AES_GCM_192((short)0x0101),
    AES_GCM_256((short)0x0102),
    CHACHA_20((short)0x0200);

    private final short value;
    SymmCryptoType(short value) {
        this.value = value;
    }

    public static SymmCryptoType fromValue(short value) {
        for (SymmCryptoType t : values())
            if (t.value == value)
                return t;
        throw new IllegalArgumentException("Unknown SymmCryptoType value: " + value);
    }
}

File: goofy-protocol-fis-fe/src/libs/goofy-protocol-core/crypto/isolated/symm/symm-crypto-type.mjs
export const SymmCryptoType = Object.freeze({
    AES_GCM_128: { name: "AES_GCM_128", value: 0x0100 },
    AES_GCM_192: { name: "AES_GCM_192", value: 0x0101 },
    AES_GCM_256: { name: "AES_GCM_256", value: 0x0102 },
    CHACHA_20: { name: "CHACHA_20", value: 0x0200 },

    fromValue(value) {
        const keys = Object.keys(this).filter(k => k !== "fromValue");
        for (const k of keys) {
            const t = this[k];
            if (t.value === value) return t;
        }
        throw new Error(`Unknown SymmCryptoType value: ${value}`);
    }
});